### PR TITLE
Feat/KFS-1173 unset compute pool

### PR DESCRIPTION
**Pull request by Guttz**
Release Notes
-------------

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok
   
What
----

Added `confluent flink compute-pool unset` for resetting default compute pool

References
----------

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/flink/command_compute_pool_unset.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	if err := c.Context.SetCurrentFlinkComputePool(""); err != nil {
+		return err
+	}
+	if err := c.Config.Save(); err != nil {
+		return err
+	}
+
+	output.Printf(errors.UnsetResourceMsg, resource.FlinkComputePool, computePoolToUnset)
```
Comment (swist) [inline internal/flink/command_compute_pool_unset.go:39]: Can you add a snapshot test for this too

**Review by swist — CHANGES_REQUESTED** [review]

File: internal/flink/command_compute_pool.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ func (c *command) newComputePoolCommand(cfg *config.Config) *cobra.Command {
 	cmd.AddCommand(c.newComputePoolListCommand())
 	cmd.AddCommand(c.newComputePoolUpdateCommand())
 	cmd.AddCommand(c.newComputePoolUseCommand())
+	cmd.AddCommand(c.newComputePoolUnsetCommand())
```
Comment (brianstrauch) [inline internal/flink/command_compute_pool.go:34]: Alphabetize!

File: test/flink_test.go [hunk 3]
```diff
@@ -29,6 +29,19 @@ func (s *CLITestSuite) TestFlinkComputePoolDelete() {
 	}
 }
 
+func (s *CLITestSuite) TestFlinkComputePoolUnset() {
+	tests := []CLITest{
+		{args: "flink compute-pool unset", login: "cloud", fixture: "flink/compute-pool/unset_before_use.golden"},
```
Comment (brianstrauch) [inline test/flink_test.go:34]:
```suggestion
		{args: "flink compute-pool unset", login: "cloud", fixture: "flink/compute-pool/unset-before-use.golden"},
```

**Comment by Guttz** [comment]
The reason we want this is that now the user has no way of stopping filtering the `confluent flink statement list` response once he set the compute pool once. After setting the compute pool once, he will always get a filtered list of statements.

File: internal/flink/command_compute_pool_unset.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/resource"
+)
+
+func (c *command) newComputePoolUnsetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unset",
+		Short: "Unset the current Flink compute pool.",
+		Long:  "Unset the current Flink compute pool that was set with `confluent Flink compute-pool use`.",
```
Comment (brianstrauch) [inline internal/flink/command_compute_pool_unset.go:16]:
```suggestion
		Long:  "Unset the current Flink compute pool that was set with `confluent flink compute-pool use`.",
```

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**